2019/fr/src/0x11-t10.md

```diff
@@ -0,0 +1,15 @@
+OWASP Top 10 Risques de sécurité des API – 2019
+===============================================
+
+| Risque | Description |
+| ------ | ----------- |
+| API1:2019 - Broken Object Level Authorization | Les API ont tendance à exposer des points d'accès qui gèrent les identifiants des objets, créant une large surface d'attaque des contrôles de niveaux d'accès. Des contrôles d'autorisation doivent être effectués pour toute fonction qui accède à une source de données à partir d'entrées fournies par un utilisateur. |
+| API2:2019 - Broken User Authentication | Les mécanismes d'authentification sont souvent implémentés incorrectement, permettant à des attaquants de compromettre des tokens d'authentification ou d'exploiter des failles d'implémentation afin d'endosser temporairement ou de manière permanente l'identité d'autres utilisateurs. La compromission de la capacité du système à identifier le client / utilisateur compromet la sécurité de l'API dans son ensemble. |
```

Comment:
```suggestion
| API2:2019 - Broken User Authentication | Les mécanismes d'authentification sont souvent implantés incorrectement, permettant à des attaquants d'obtenir des jetons (tokens) d'authentification ou d'exploiter des failles issues de défauts d'implantation logiciel afin d'endosser temporairement ou de manière permanente l'identité d'autres utilisateurs. La compromission de la capacité du système à identifier le client / utilisateur compromet la sécurité de l'API dans son ensemble. |
```

Reply:
Je sais que le terme compromettre à une saveur très particulière en IT Sec mais je pense que dans ce contexte le verbe se rapproche plus du sens d'exposer / mettre à nue ou/et donc obtenir.

Reply:
La notion de compromission dépasse l'utilisation à l'identique d'un jeton, elle inclut la possible manipulation d'un ou plusieurs de ses composants.

La notion d'implémentation renvoie à la conception/réalisation d'un projet (ici d'une API), celle d'implantation renvoyant à l'installation/configuration d'un logiciel.

Les termes compromission et implémentation ont été pareillement employés dans la version française de l'[OWASP Top 10 Risques Sécurités des Applications 2017](https://github.com/OWASP/Top10/blob/master/2017/fr/0x11-t10.md) (cf point A2).

Reply:
Personnellement je suis friand d'anglicismes donc cela ne me dérange pas plus. Mais tout de même, il semblerait qu'en français on parle bien d'implantation concernant la conception/réalisation de projet informatique (cf l'[académie](https://www.academie-francaise.fr/implementer)). J'ai même le souvenir d'utiliser le verbe "implémenter" à l'oral comme à l'écrit depuis très longtemps et d'avoir été sujet à de nombreuses remarques, notamment à l'école, mais passons.

Reply:
Comprenez bien, que même s'il est d'usage courant de parler d'implémentation en informatique. D'implémenter ça ou ça lorsque l'on s'adresse aux collègues ou bien aux clients, que cela nous semble à tous (IT guys) totalement nautrel, ça ne rend pas l'expression plus française et devrait donc être proscrite autant que faire se peut à l'écrit.

En ce qui concerne, "compromission", "compromettre un jeton d'auth." ce mot est parti intégrante du vocabulaire attribué à la sécurité informatique et à la sécurité en générale et donc son utilisation ici me dérange beaucoup moins. Je pensais que ma proposition, "d'obtenir" pour un attaquant d'une manière ou d'une autre, rendait le processus plus clair mais vous m'avez confirmé que non.